src/highdicom/sr/templates.py

```diff
@@ -4415,27 +4809,41 @@ def __init__(
             ),
             relationship_type=RelationshipTypeValues.CONTAINS
         )
-        content = ContentSequence()
-        if groups is not None:
-            for descriptor_items in groups:
-                group_item = ContainerContentItem(
+        library_item.ContentSequence = ContentSequence()
+        if datasets is not None:
+            # We will use just one group
```

Comment:
> In your last comment - did you mean Frame of Reference UID instead of Transfer Syntax UID?

Yes, sorry I meant Frame of Reference UID.

> In this case - should we create groups by modality/frame of reference where these are available - and otherwise fall back on only modality to create the groups?

Good point! Yes, that makes sense.

Reply:
Another use case for this is grouping images by specific series. This is the example we put together to accompany the prostate reporting templates: http://dicom.nema.org/medical/dicom/current/output/chtml/part17/sect_WWWW.2.html.

Reply:
The code now tests images to see if they contain a FrameOfReferenceUID; if they do not only the Modality is used for creating groups.  For a future PR we might pass a function that generated different partitions or provide an option to specify an ImageLibrary as an argument.